src/screens/dialogs/move-file-dialog.tsx

```diff
@@ -0,0 +1,154 @@
+  const directory = searchParams.get("directory");
+
+  useEffect(() => {
+    if (Open && selectedFile) {
+      const fetchFolders = async () => {
+        const { success, ...res } = await listFilesService({
+          token: session?.token as string,
+          directory: selectedFolder ? selectedFolder.uuid : directory
+        });
+
+        if (!success) {
+          toast.error(res.msg);
+        } else {
+          const folderList = res.files.filter(
+            (file) => !file.isFile && file.uuid !== selectedFile?.uuid
+          );
+          setFolders(folderList);
+        }
+      };
+
+      fetchFolders();
+    }
+  }, [directory]);
+
+  const handleMove = async () => {
+    if (moving || !selectedFolder || !selectedFile) return;
+
+    if (!selectedFolder.isFile) {
+      setMoving(true);
+      const { success, ...res } = await moveFileService({
+        directoryUUID: selectedFolder.uuid,
+        fileUUID: selectedFile.uuid as string,
+        token: session?.token as string
+      });
+
+      if (!success) {
+        toast.error(res.msg);
+        setMoving(false);
+        return;
+      }
+
+      const successMessage = selectedFile.isFile
+        ? "File has been moved successfully"
+        : "Folder has been moved successfully";
+
+      toast.success(successMessage);
+      setMoving(false);
+      closeDialog(AVAILABLE_DIALOGS.MOVE_FILE);
+      setSelectedFolder(null);
+      removeFile(selectedFile.uuid);
+    } else {
+      toast.error("This action cannot be performed");
+    }
+  };
+
+  const changeDirectory = (directoryUUID: string) => {
+    if (!selectedFile) return;
+
+    const selectedFolder = folders.find(
+      (folder) => folder.uuid === directoryUUID
+    );
+    if (!selectedFolder) {
+      return;
+    }
+
+    setSelectedFolder(selectedFolder);
+    setFolders([]);
+
+    loadFolderContents(directoryUUID);
+  };
+
+  const loadFolderContents = async (directoryUUID: string) => {
+    const { success, ...res } = await listFilesService({
+      token: session?.token as string,
+      directory: directoryUUID
+    });
+
+    if (success) {
+      const contents = res.files.filter(
+        (file) => file.uuid !== selectedFile?.uuid
+      );
+      setFolders(contents);
+    } else {
+      toast.error(res.msg);
+    }
+  };
+
+  return (
+    <Dialog
+      isOpen={Open}
+      onClose={() => {
+        closeDialog(AVAILABLE_DIALOGS.MOVE_FILE);
+        setSelectedFolder(null);
+        setFolders([]);
+      }}
+      title="Move File"
+    >
+      <div className="max-h-[300px] overflow-y-auto p-4">
+        <div className="navigation-panel">
+          {folders.map((folder) => (
+            <button
+              key={folder.uuid}
+              className="hover-bg-blue-700 mb-2 h-12 w-full rounded-lg bg-blue-600 px-4 py-2 text-left"
+              onClick={() => changeDirectory(folder.uuid)}
+              aria-label={`Dialog ${folder.name}`}
```

Comment:
The label is not sufficiently descriptive. I think something like "Move to ${folder.name}" would be more appropriate. Please consider a better label.